### DISCUSSION TITLE
About C++ interface

### DISCUSSION THREAD

**Discussion by hu-hamster** [deepmodeling/deepmd-kit#1428] [Q&A]
When I run the program using the C++ interface, the error is displayed as follows
![image](https://user-images.githubusercontent.com/61817777/149929257-d2dae9fe-a79d-4a0f-aa4f-5f5bfc3dc8fd.png)
The command I execute is "dp_ipi input.json"，did this mean that I failed to install the C++ interface

**Comment by njzjz** [comment]
dp_ipi is the interface for i-Pi. Do you want training? That's in the Python interface.

**Reply by hu-hamster** [reply]
Thank you for your reply,  does this mean that ASC22's improvement to the code are limited to the Python interface

**Reply by njzjz** [reply]
I don't know the exact requirement of ASC22. The architecture of DeePMD-kit code is given below:

![image](https://user-images.githubusercontent.com/9496702/150169583-01fe994a-c17b-408f-96ab-e56892936cec.png)

Note that the Python interface is based on other modules.

**Reply by hu-hamster** [reply]
Thank you very much
